### PR TITLE
.Net: KernelFunctionArguments class has been added

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
The class will be used to represent arguments for both Kernel.InvokeAsync and KernelFunction.InvokeAsync methods. 

- Signature of the KernelFunction.InvokeAsync methods will change from the current one:
_InvokeAsync(Kernel kernel, **SKContext context, AIRequestSettings? requestSettings,** CancellationToken cancellationToken)_
to the new one:
_InvokeAsync(Kernel kernel, **KernelFunctionArguments arguments**, CancellationToken cancellationToken)_

- Similarly, signature of the Kernel.InvokeAsync methods will change from the current one:
_InvokeAsync(this Kernel kernel, KernelFunction function, **ContextVariables variables**, CancellationToken cancellationToken)_ 
to the new one:
_InvokeAsync(this Kernel kernel, KernelFunction function, **KernelFunctionArguments arguments**,  CancellationToken cancellationToken)_ 

Currently, SK works with string arguments 99% of the time. In the future, the argument value type will change from string to object? in scope of the complex-type support feature.
 

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+/// <summary>
+/// Represents arguments of <see cref="KernelFunction.InvokeAsync"/> and "Kernel.InvokeAsync" methods.
+/// </summary>
+public sealed class KernelFunctionArguments : Dictionary<string, string>
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KernelFunctionArguments"/> class.
+    /// </summary>
+    /// <param name="requestSettings">The AI request settings.</param>
+    public KernelFunctionArguments(AIRequestSettings? requestSettings = null)
```
Comment (stephentoub) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:19]: This should initialize the base with OrdinalIgnoreCase, right?  Or are we switching to make argument matching case-sensitive?

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#pragma warning disable IDE0130
+namespace Microsoft.SemanticKernel;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Represents arguments of <see cref="KernelFunction.InvokeAsync"/> and "Kernel.InvokeAsync" methods.
+/// </summary>
+public sealed class KernelFunctionArguments : Dictionary<string, string>
+{
```
Comment (stephentoub) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:14]: I suggest there should also be a constructor that accepts a KernelFunctionArguments and another that accepts an `IDictionary<>`.

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:1]: QQ, do we have plans in doing this a Dictionary<string, object> in a future iteration?

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:19]: Right, completely forgot about this aspect. Fixed.

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:14]: I had both in my POC but wanted to add them along the way if they are needed. Anyway, they are added.

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs
Comment (SergeyMenshykh) [reply dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:1]: Yep, we have plans to support complex types, so we'll change the value type to - Dictionary<string, object?>

**Review by RogerBarreto — APPROVED** [review]

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    /// <param name="other">The <see cref="KernelFunctionArguments"/> whose AI request setting and elements are copied to the new <see cref="KernelFunctionArguments"/>.</param>
+    public KernelFunctionArguments(KernelFunctionArguments other) : this(other as IDictionary<string, string>)
+    {
+        this.RequestSettings = other.RequestSettings;
+    }
+
+    /// <summary>
+    /// Gets or sets the AI request settings.
+    /// </summary>
+    public AIRequestSettings? RequestSettings { get; set; }
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:46]: `public PromptExecutionSettings? ExecutionSettings { get; set; }`

File: dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    /// Initializes a new instance of the <see cref="KernelFunctionArguments"/> class that contains AI request settings and elements copied from the specified <see cref="KernelFunctionArguments"/>
+    /// </summary>
+    /// <param name="other">The <see cref="KernelFunctionArguments"/> whose AI request setting and elements are copied to the new <see cref="KernelFunctionArguments"/>.</param>
+    public KernelFunctionArguments(KernelFunctionArguments other) : this(other as IDictionary<string, string>)
+    {
+        this.RequestSettings = other.RequestSettings;
+    }
+
+    /// <summary>
+    /// Gets or sets the AI request settings.
```
Comment (markwallace-microsoft) [inline dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionArguments.cs:44]: Gets or sets the prompt execution settings

**Review by markwallace-microsoft — COMMENTED** [review]
There is a rename needed

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**